x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlRealm.java

```diff
@@ -155,6 +158,8 @@ public final class SamlRealm extends Realm implements Releasable {
     // Although we only use this for IDP metadata loading, the SSLServer only loads configurations where "ssl." is a top-level element
     // in the realm group configuration, so it has to have this name.
 
+    private static final AtomicBoolean REFRESHING_METADATA = new AtomicBoolean(false);
```

Comment:
I'd love to make it an instance level field, but it's a messy refactoring - all the code that loads metadata is static.
It would be good to clean that up (I think moving the metadata resolution into a separate class would be ideal) but I don't think it can be tackled in this PR.
Let me know if you feel strongly that it needs to change.

Reply:
I'd be happy with just a comment explaining the choice here. Thanks!